03-rnaseq/clustering/clustering_heatmap_example.Rmd

```diff
@@ -0,0 +1,334 @@
+
+# Subset the data choosing only genes whose variances are in the upper quartile
+df_by_var <- data.frame(assay(dds_norm)) %>%
+  dplyr::filter(variances > upper_var)
+```
+
+### Create a heatmap
+To further customize the heatmap, see a vignette for a guide at this [link](https://slowkow.com/notes/pheatmap-tutorial/).
+
+```{r}
+# Create and store the technical replicates heatmap object
+pheatmap <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE, # We want to cluster the heatmap by rows (genes in this case)
+    cluster_cols = TRUE, # We also want to cluster the heatmap by columns (samples in this case)
+    main = "Clustered Heatmap",
+    colorRampPalette(c(
+      "deepskyblue",
+      "black",
+      "yellow"
+    ))(25
+    ),
+    scale = "row" # Scale values in the direction of genes (rows)
+  )
+```
+
+We've created a heatmap but although our genes and samples are clustered, there
+is not much information that we can gather here because we did not provide the
+`pheatmap` function with annotation labels for our samples.
+
+First let's save our clustered heatmap.
+
+#### Save heatmap as a png
+You can easily switch this to save to a jpeg or tiff by changing the function 
+and file name within the function to the respective file suffix.
+
+```{r}
+# Open a png file
+png(file.path(
+  plots_dir,
+  "HeatmapSRP070849_clustered.png" # Replace file name with a relevant output plot name
+))
+
+# Print your heatmap
+pheatmap
+
+# Close the png file:
+dev.off()
+```
+
+Now, let's add some annotation bars to our heatmap.
+
+### Prepare metadata for annotation
+We are going to manipulate the metadata and add variables with the information
+that we would like to use to annotate the heatmap.
+
+```{r}
+# Let's prepare the annotation data.frame for the uncollapsed `DESeqData` set
+# object which will be used to create the technical replicates heatmap
+annotation_df <- metadata %>%
+  # Create a variable to store the cancer type information
+  dplyr::mutate(
+    cancer_type = dplyr::case_when(
+      startsWith(refinebio_title, "TET2") ~ "TET2",
+      startsWith(refinebio_title, "IDH2") ~ "IDH2",
+      startsWith(refinebio_title, "WT") ~ "WT",
+      TRUE ~ "unknown"
+    )
+  ) %>%
+  # We want to select the variables that we want for annotating the technical
+  # replicates heatmap
+  dplyr::select(
+    refinebio_accession_code,
+    cancer_type,
+    refinebio_treatment
+  ) %>%
+  # The `pheatmap` function requires that the rownames of our annotation object
+  # matches the column names of our `DESeaDataSet` object
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+#### Create annotated heatmap
+You can create an annotated heatmap by providing our annotation object to the
+`annotation_col` argument of the `pheatmap` function.
+
+```{r}
+# Create and store the annotated heatmap object
+pheatmap_annotated <-
+  pheatmap(
+    df_by_var,
+    cluster_rows = TRUE,
+    cluster_cols = TRUE,
```

Comment:
We need to add a `show_rownames = FALSE` on both heatmaps. The rownames are a bit smooshy and not really necessary.